src/app/components/shared/map/map.component.ts

```diff
@@ -72,31 +56,25 @@ export class MapComponent
   // Setting to "hybrid" can increase load times and looks like the map is bugged
   public mapOptions: MapOptions = { mapTypeId: "satellite" };
   public bounds: google.maps.LatLngBounds;
-  private updateMap: boolean;
+  private updateMap: boolean = false;
```

Comment:
what is the purpose of this field, what does it do?

Reply:
Google maps runs the `ngAfterViewChecked` method every 100ms, even with `DetectChanges.OnPush`

This `updateMap` exists so that we don't need to run this code every 100ms
![image](https://github.com/user-attachments/assets/192cd0e1-831c-469f-8dc9-51bcfbdbc3a1)

I'm looking into it now because I suspect this is something that we're doing wrong; I wouldn't expect Google to author code that continuously runs `ngAfterViewChecked`

Reply:
oh i didn't see that function.

you're right though, seems weird we're hooked into ngAfterViewChecked. Perhaps another hook is more appropriate?

Reply:
Update: `ngAfterViewChecked` is due to the `ViewChild` decorators for the `GoogleMap`, `MapInfoWindow`, and `MapMarker` components